externs/es6.js

```diff
@@ -1093,6 +1093,26 @@ Array.prototype.keys;
 Array.prototype.entries;
 
 
+/**
+ * @param {?function(this:S, T, number, !Array<T>): boolean} predicate
```

Comment:
Can we make this non-null? It doesn't look like there's a use case for passing null here.